util/fipstools/capture_hash/capture_hash.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// capture_hash runs another executable that has been linked with libcrypto. It expects the libcrypto to run the power
+// on self tests and fail due to a fingerprint mismatch. capture_hash parses the output, takes the correct fingerprint
+// value, and generates a new C file that contains the correct fingerprint which is used to build the final libcrypto.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const line0 = "FIPS integrity test failed."
+
+// This must match what is in crypto/fipsmodule/fips_shared_support.c
+const line1 = "Expected:   ae2cea2abda6f3ec977f9bf6949afc836827cba0a09f6b6fde52cde2cdff3180"
+const hash_len = 64
+func main() {
+	executable := flag.String("in-executable", "", "Path to the executable file")
+	flag.Parse()
+	cmd := exec.Command(*executable)
+	out, err := cmd.CombinedOutput()
+	if err == nil {
+		fmt.Fprintf(os.Stderr, string(out))
+		panic("Executable did not fail as expected")
+	}
+	lines := strings.Split(string(out), "\r\n")
+	if len(lines) != 4 {
+		fmt.Fprintf(os.Stderr, string(out))
+		panic(fmt.Sprintf("Expected 4 lines in output but got %d", len(lines)))
+	}
+	num_lines := len(lines)
+	if lines[num_lines-4] != line0 {
```

Comment:
Why the condition has lines[num_lines-j] and the printed error has lines[i], j descending from 4 and i ascending from 0?

Reply:
Yeah I don't know what I was thinking here. I'll simplify all of it.

Reply:
I think I was trying to be clever and was printing debug info in the windows start up code at some point, so I was trying to support more lines printed before the BCM error message. Now it looks confusing and we don't need it.